chia/plotting/create_plots.py

```diff
@@ -19,56 +20,145 @@
+        if not self.resolved:
+            raise Exception("resolve() hasn't been called yet")
+        return self._farmer_public_key_g1element
+
+    @property
+    def pool_public_key_g1element(self):
+        if not self.resolved:
+            raise Exception("resolve() hasn't been called yet")
+        return self._pool_public_key_g1element
+
+    @property
+    def pool_contract_puzzle_hash(self):
+        if not self.resolved:
+            raise Exception("resolve() hasn't been called yet")
+        return self._pool_contract_puzzle_hash
+
+    async def get_sk(self, keychain_proxy: Optional[KeychainProxy] = None) -> Optional[Tuple[PrivateKey, bytes]]:
+        sk: Optional[PrivateKey] = None
+        if keychain_proxy:
+            try:
+                if self.alt_fingerprint is not None:
+                    sk = await keychain_proxy.get_key_for_fingerprint(self.alt_fingerprint)
+                else:
+                    sk = await keychain_proxy.get_first_private_key()
+            except Exception as e:
+                log.error(f"Keychain proxy failed with error: {e}")
+        else:
+            sk_ent: Optional[Tuple[PrivateKey, bytes]] = None
+            keychain: Keychain = Keychain()
+            if self.alt_fingerprint is not None:
+                sk_ent = keychain.get_private_key_by_fingerprint(self.alt_fingerprint)
+            else:
+                sk_ent = keychain.get_first_private_key()
+
+            if sk_ent:
+                sk = sk_ent[0]
+        return sk
+
+    async def get_farmer_public_key(self, keychain_proxy: Optional[KeychainProxy] = None) -> G1Element:
+        sk: Optional[PrivateKey] = await self.get_sk(keychain_proxy)
+        if sk is None:
+            raise RuntimeError(
+                "No keys, please run 'chia keys add', 'chia keys generate' or provide a public key with -f"
+            )
+        return master_sk_to_farmer_sk(sk).get_g1()
 
+    async def get_pool_public_key(self, keychain_proxy: Optional[KeychainProxy] = None) -> G1Element:
+        sk: Optional[PrivateKey] = await self.get_sk(keychain_proxy)
+        if sk is None:
+            raise RuntimeError(
+                "No keys, please run 'chia keys add', 'chia keys generate' or provide a public key with -p"
+            )
+        return master_sk_to_pool_sk(sk).get_g1()
 
-def get_pool_public_key(alt_fingerprint: Optional[int] = None) -> G1Element:
-    sk_ent: Optional[Tuple[PrivateKey, bytes]]
-    keychain: Keychain = Keychain()
-    if alt_fingerprint is not None:
-        sk_ent = keychain.get_private_key_by_fingerprint(alt_fingerprint)
-    else:
-        sk_ent = keychain.get_first_private_key()
-    if sk_ent is None:
-        raise RuntimeError("No keys, please run 'chia keys add', 'chia keys generate' or provide a public key with -p")
-    return master_sk_to_pool_sk(sk_ent[0]).get_g1()
 
+async def create_plots(args, root_path, use_datetime=True, test_private_keys: Optional[List] = None):
```

Comment:
Is it possible to call `keys_resolver.resolve()` outside `create_plots`, and pass the result of that in as a bundle of key-specific data?

That would allow us to have tests for create_plots that use dummy key data, and not tie it to the method we use to get the keys.